frontend/app.js

```diff
@@ -0,0 +1,510 @@
+var React = require('react');
+var ReactDOM = require('react-dom');
+
+class Root extends React.Component {
+  constructor() {
+    super()
+
+    this.state = {
+      tasks: null,
+      taskName: '',
+      taskDescription: '',
+      childVisible: false,
+      submitBar: false,
+      completedState: 0
+    };
+  }
+
+  componentDidMount() {
+    fetch('/task')
+      .then((data) => {
+        return data.json()
+      })
+      .then((json) => {
+        this.setState({
+          tasks: json.tasks
+        })
+      })
+  }
+
+  addTask(event) {
+    event.preventDefault();
+    var name = this.state.taskName
+    var description = this.state.taskDescription
+    if (name == '') {
+      alert('you must write something')
+    } else {
+
+      fetch('/task/', {
+        method: 'POST',
+        headers: {
+          'Accept': 'application/json',
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          name: name,
+          description: description
+        })
+      }).then((data) => {
+          return data.json()
+        })
+        .then((json) => {
+
+          this.setState({
+            childVisible: false,
+            tasks: json.tasks,
+            taskName: '',
+            taskDescription: '',
+            submitBar: false
+          })
+
+      })
+    }
+  }
+
+  removeTask(event) {
+    fetch('/task/', {
+      method: 'DELETE',
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        id: event
+      })
+    }).then((data) => {
+        return data.json()
+      })
+      .then((json) => {
+        this.setState({
+          tasks: json.tasks
+        })
+      })
+  }
+
+  completeTask(event) {
+    console.log(event)
+    fetch('/task/', {
+      method: 'PUT',
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        id: event
+      })
+    }).then((data) => {
+        return data.json()
+      })
+      .then((json) => {
+        this.setState({
+          tasks: json.tasks
+        })
+      })
+  }
+
+  onTaskChange(event) {
+    this.setState({
+      taskName: event.target.value,
+    });
+    if (event.target.value.length > 0) {
+      this.setState({
+        childVisible: true
+      })
+    } else {
+      this.setState({
+        submitBar: false,
+        childVisible: false,
+        taskDescription: ''
+      })
+    }
+  }
+
+  onDesciptionChange(event) {
```

Comment:
I'd merge the `setState` calls into one. They actually get batched anyway, so it doesn't make a real difference other than asthetics.

```javascript
const { value } = event.target;

this.setState({
  taskDescription: value,
  submitBar: value.length > 0
});
```